src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -571,7 +556,7 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
             // simplicity).
             if (entryChange.readOnly)
             {
-                metrics.mMetadataSizeByte +=
+                metrics.mContractEventSizeByte +=
```

Comment:
Shouldn't we remove this? Or is this a hack to charge for the expiration extension write?

Reply:
Nice catch, removed.

Reply:
One unfortunate side effect of moving from meta bytes to event bytes is that we won't charge for expiration extension writes at all.

Reply:
I don't think this is the right place to charge for this anyway (metadata is much cheaper than ledger writes).

If we want this to be completely transparent, then probably not charging anything is the right thing. However, I think it might be safer to do something. I see a few options:

- Add expiration bumps to `writeBytes`
- Add expiration bumps to `writeBytes` *and* RW entry charges. While this is somewhat harder to convey to the users, it seems like the safest options (otherwise we need to always be readly to write RO+RW entries to the bucket list). This is tricky though, as bumps are not a part of the footprint. Basically we would probably need to count the total number of entries in the footprint towards the ledger writes limits.
- Add a flat factor to the rent charge formula. We need some math to figure it out, but I think that might work as well. The benefit is that bumps would look like read-only from the resources standpoint.